src/lib/crypto/kimchi_bindings/wasm/src/srs.rs

```diff
@@ -261,8 +253,8 @@ pub mod fp {
         domain_size: i32,
         i: i32,
     ) -> Option<WasmPolyComm> {
-        let bases = srs.0.lagrange_bases.get(&(domain_size as usize));
-        bases.map(|bases| bases[i as usize].clone().into())
+        let basis = srs.get_lagrange_basis_from_domain_size(domain_size as usize);
+        Some(basis[i as usize].clone().into())
```

Comment:
@45930 @mrmr1993 this change needs to be undone to unbreak Lagrange Basis caching and fix the performance regression. There was a reason this returned an option and wasn't simply doing the expensive computation